PaddleSlim/README.md

```diff
@@ -0,0 +1,176 @@
+
+<div align="center">
+  <h3>
+    <a href="docs/tutorial.md">
+      算法原理介绍
+    </a>
+    <span> | </span>
+    <a href="docs/usage.md">
+      使用文档
+    </a>
+    <span> | </span>
+    <a href="docs/demo.md">
+      示例文档
+    </a>
+    <span> | </span>
+    <a href="docs/model_zoo.md">
+      Model Zoo
+    </a>
+  </h3>
+</div>
+
+
+---
+# PaddleSlim模型压缩工具库
+
+PaddleSlim是PaddlePaddle框架的一个子模块。PaddleSlim首次在PaddlePaddle 1.4版本中发布。在PaddleSlim中，实现了目前主流的网络剪枝、参数量化、模型蒸馏三种压缩策略，主要用于压缩图像领域模型。在后续版本中，会添加更多的压缩策略，以及完善对NLP领域模型的支持。
+
+## 目录
+- [特色](#特色)
+- [架构介绍](#架构介绍)
+- [功能列表](#功能列表)
+- [实验结果与ModelZoo](#简要实验结果)
+- [模型导出格式](#模型导出格式)
+
+## 主要特点
+
+Paddle-Slim工具库有以下特点：
+
+###  接口简单
+
+- 以配置文件方式集中管理可配参数，方便实验管理
+- 在普通模型训练脚本上，添加极少代码即可完成模型压缩
+
+详见：[使用示例](docs/demo.md)
+
+### 效果好
+
+- 对于冗余信息较少的MobileNetV1模型，卷积核剪切策略依然可缩减模型大小，并保持尽量少的精度损失。
+- 蒸馏压缩策略可明显提升原始模型的精度。
+- 量化训练与蒸馏的组合使用，可同时做到缩减模型大小和提升模型精度。
+
+详见：[效果数据与ModelZoo](docs/model_zoo.md)
+
+### 功能更强更灵活
+
+- 剪切压缩过程自动化
+- 剪切压缩策略支持更多网络结构
+- 蒸馏支持多种方式，用户可自定义组合loss
+- 支持快速配置多种压缩策略组合使用
+
+详见：[使用说明](docs/usage.md)
+
+## 架构介绍
+
+这里简要介绍模型压缩工具实现的整体原理，便于理解使用流程。
+**图 1**为模型压缩工具的架构图，从上到下为API依赖关系。蒸馏模块、量化模块和剪切模块都间接依赖底层的paddle框架。考虑到这种依赖关系和版本的管理，我们将模型压缩工具作为了paddle框架的一部分，所以已经安装普通版本paddle的用户需要重新下载安装支持模型压缩功能的paddle，才能使用压缩功能。
+
+<p align="center">
+<img src="docs/images/framework_0.png" height=452 width=706 hspace='10'/> <br />
+<strong>图 1</strong>
+</p>
+
+如**图 1**所示，最上层的紫色模块为用户接口，在Python脚本中调用模型压缩功能时，只需要构造一个Compressor对象即可，在[使用文档](docs/usage.md)中会有详细说明。
+
+我们将每个压缩算法称为压缩策略，在迭代训练模型的过程中调用用户注册的压缩策略完成模型压缩，如**图2**所示。其中，模型压缩工具封装好了模型训练逻辑，用户只需要提供训练模型需要的网络结构、数据、优化策略（optimizer）等，在[使用文档](docs/usage.md)会对此详细介绍。
+
+<p align="center">
+<img src="docs/images/framework_1.png" height=255 width=646 hspace='10'/> <br />
+<strong>图 2</strong>
+</p>
+
+## 功能列表
+
+
+### 剪切
+
+- 支持敏感度和uniform两种方式
+- 支持VGG、ResNet、MobileNet等各种类型的网络
+- 支持用户自定义剪切范围
+
+### 量化训练
+
+- 支持动态和静态两种量化训练方式
+  - 动态策略: 在每次迭代过程中动态统计并使用量化参数。
+  - 静态策略: 对不同的输入，采用相同的从训练数据中统计得到的量化参数。
+- 支持对权重全局量化和Channel-Wise量化
+- 支持以兼容Paddle Mobile的格式保存模型
+
+### 蒸馏
+
+- 支持在teacher网络和student网络任意层添加组合loss
+  - 支持FSP loss
+  - 支持L2 loss
+  - 支持softmax with cross-entropy loss
+
+### 其它功能
+
+- 支持配置文件管理压缩任务超参数
+- 支持多种压缩策略组合使用
+- 蒸馏和剪切压缩过程支持checkpoints功能
+
+## 简要实验结果
+
+本节列出了PaddleSlim模型压缩工具库的一些实验结果，更多实验数据和预训练模型的下载，请参考：[详细实验结果与ModelZoo](docs/model_zoo.md)
+
+### 量化训练
+
+评估实验所使用数据集为ImageNet1000类数据，且以top-1准确率为衡量指标：
+
+| Model | FP32| int8(A:abs_max, W:abs_max) | int8, (A:moving_average_abs_max, W:abs_max) |int8, (A:abs_max, W:channel_wise_abs_max) |
+|:---|:---:|:---:|:---:|:---:|
+|MobileNetV1|70.916%|71.008%|70.84%|71.00%|
+|ResNet50|76.352%|76.612%|76.456%|76.73%|
```

Comment:
Please update the quantization result, refer to http://agroup.baidu.com/paddlecv/md/article/1642698.